module/VuFind/src/VuFind/AjaxHandler/GetResultCount.php

```diff
@@ -0,0 +1,105 @@
+<?php
+/**
+ * Ajax Controller for Libraries Extension
+ *
+ * PHP version 5
+ *
+ * Copyright (C) Staats- und Universitätsbibliothek 2017.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ *
+ * @category VuFind2
+ * @package  Controller
+ * @author   Hajo Seng <hajo.seng@sub.uni-hamburg.de>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://github.com/subhh/beluga
+ */
+namespace VuFind\AjaxHandler;
+
+use Laminas\Mvc\Controller\Plugin\Params;
+use Laminas\Stdlib\Parameters;
+use VuFind\Search\Results\PluginManager as ResultsManager;
+
+/**
+ * This controller handles global AJAX functionality
+ *
+ * @category VuFind2
+ * @package  Controller
+ * @author   Chris Hallberg <challber@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     http://vufind.org/wiki/vufind2:building_a_controller Wiki
+ */
+class GetResultCount extends AbstractBase
+{
+    /**
+     * ResultsManager
+     *
+     * @var resultsManager
+     */
+    protected $resultsManager;
+
+    /**
+     * Constructor
+     *
+     * @param ResultsManager $resultsManager Results Manager
+     */
+    public function __construct(ResultsManager $resultsManager)
+    {
+        $this->resultsManager = $resultsManager;
+    }
+
+    /**
+     * Handle a request.
+     *
+     * @param Params $params Parameter helper from controller
+     *
+     * @return array [response data, HTTP status code]
+     */
+    public function handleRequest(Params $params)
+    {
+        $queryString = $params->fromQuery('querystring');
+        $queryString = urldecode(
+            str_replace(
```

Comment:
Why is this str_replace necessary? Is the Javascript code sending things entity-encoded? If so, is it possible that you can fix that on the JS side using the `text()` method or something similar?

Reply:
To be true, I haven't understood the encodings use by the communication between javascript and backend. I'm not so much used in javascript anyway. But I'll try it.